### DISCUSSION TITLE
Feature request: Eufy support

### DISCUSSION THREAD

**Discussion by nukithelegend** [koush/scrypted#627] [Q&A]
Hi

Reviving this again since I see there is actually a Eufy folder in plugins, but will Eufy cameras (like the 2C Pro) be supported?

I can't seem to install the eufy plugin, even though it shows in the plugins folder.

**Comment by mlavio829** [comment]
Would love for this to work as well

**Comment by leedsalex** [comment]
I tested with the battery camera and it's a terrible experience, I don't see support being added or further development being done for Eufy w/ scrypted... unless someone else picks up where I left off. I already returned my Eufy camera. Sorry!

**Comment by Haasie** [comment]
Eufy also sell cameras with power plug ( pan, tilt,zoom for instance). I have one of those and I must say, it works really nice with their app. Setting up the RTSP stream in to scrypted generates a lot of artefacts. If there I a solution for it, I would be really happy if I know how.

**Comment by sunshineV9** [comment]
I would be open up to take on the eufy security plugin. I for myself own the pan and tilt wifi camera.
Hopefully I have some time this or next week to setup my dev environment and take look onto it.

**Reply by sunshineV9** [reply]
Today I finally got some time, to look into this. Unfortunately I think this wont be possible without support. I tried using the eufy-security-client (https://github.com/bropat/eufy-security-client) package, which already reverse engineered everything we should need for a scrypted integration, but they are using some dependencies which are not compatible with scrypted, e.g. got (https://www.npmjs.com/package/got) which is not compatible with webpack, afaik.
Currently I do not have the time to investigate this further, but I am open up for any help on this.

If anyone is interested, i created a plugin repo here: https://github.com/sunshineV9/eufy-security-scrypted

**Comment by koush** [comment]
put got into optionalDependencies and it will web pack fine. There's also an incomplete plugin implementation (streaming works) in the scrypted repo.

**Reply by sunshineV9** [reply]
Thanks for you reply! I will try it out.

Yes, I saw that, but I wasn't sure if I could just take it over. So I created my own repo so I can't mess anything up here 😅

**Comment by sunshineV9** [comment]
made some progress today, I am now able to connect with the eufy-security-client and implemented basic discovery of devices, 
on/off switch to enable/disable camera and displaying the last picture taken by the camera in scrypted. 
Currently I cant get the implementation for streaming around my head. Also looked into the existing plugin, but unfortunately I didn't get that working, because I don't know how to access other plugins from within mine.

For now I will call it a day, I will join on Discord and ask further questions there :)

**Reply by mistert871** [reply]
Anu update / progress? I am also looking for this implementation. Thanks

**Reply by JEMcats** [reply]
Hello, I currently have a P2P eufy floodlight camera with no RTSP or HomeKit support, I use homebridge-eufy-security [v2.3.0](https://github.com/homebridge-eufy-security/plugin/tree/v2.3.0) at the moment and it works well but the video stream is choppy, I would love to use scrypted instead of homebridge as the new versions of homebridge-eufy-security don't seem to work with my camera. If somebody could work on this I would be able to help test new versions of the plugin.

**Comment by JimboJamez** [comment]
Has there been any progress made on this? I am trying to integrate my EufyCam 2's into scrypted but no matter how I structure my rtsp stream I am getting "401 Unauthorized" and "Stream not found" errors. 

Examples below:

**rtsp://(IP ADDRESS)/live2:**

> [Rebroadcast Plugin]: Stream 1 prebuffer session started
> [Rebroadcast Plugin]: Stream 1 prebuffer session starting
> stream url rtsp://(IP ADDRESS)/live2
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  OPTIONS rtsp://(IP ADDRESS)/live2 RTSP/1.0
> [Rebroadcast Plugin]: CSeq: 0
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 200 OK
> [Rebroadcast Plugin]: CSeq: 0
> [Rebroadcast Plugin]: Public: OPTIONS, DESCRIBE, SETUP, TEARDOWN, PLAY, GET_PARAMETER
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  DESCRIBE rtsp://(IP ADDRESS)/live2 RTSP/1.0
> [Rebroadcast Plugin]: Accept: application/sdp
> [Rebroadcast Plugin]: CSeq: 1
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 **401 Unauthorized**
> [Rebroadcast Plugin]: CSeq: 1
> [Rebroadcast Plugin]: WWW-Authenticate: Digest realm="C7H2Dpn", nonce="cuD0Cc652Hyamsa4cw31iCb"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  DESCRIBE rtsp://(IP ADDRESS)/live2 RTSP/1.0
> [Rebroadcast Plugin]: Accept: application/sdp
> [Rebroadcast Plugin]: CSeq: 2
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: Authorization: Digest username="(USERNAME)", realm="C7H2Dpn", nonce="cuD0Cc652Hyamsa4cw31iCb", uri="rtsp://(IP ADDRESS)/live2", algorithm="MD5", response="be03e775ca4145383d24e588e7e0e6c5"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 **404 Stream Not Found**
> [Rebroadcast Plugin]: CSeq: 2
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  TEARDOWN rtsp://(IP ADDRESS)/live2 RTSP/1.0
> [Rebroadcast Plugin]: CSeq: 3
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: Authorization: Digest username="(USERNAME)", realm="C7H2Dpn", nonce="cuD0Cc652Hyamsa4cw31iCb", uri="rtsp://(IP ADDRESS)/live2", algorithm="MD5", response="88190f0d5ed78cdb93c9f481b1d7d4b7"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: Stream 1 prebuffer session ended with error M [Error]: RTSP Error: RTSP/1.0 **404 Stream Not Found**
> [Rebroadcast Plugin]:     at t.RtspClient.request (/@scrypted/prebuffer-mixin/main.nodejs.js:1:91847)
> [Rebroadcast Plugin]:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
> [Rebroadcast Plugin]:     at async t.startRtspSession (/@scrypted/prebuffer-mixin/main.nodejs.js:1:144136)
> [Rebroadcast Plugin]:     at async V.startPrebufferSession (/@scrypted/prebuffer-mixin/main.nodejs.js:1:117624) {
> [Rebroadcast Plugin]:   status: {
> [Rebroadcast Plugin]:     line: 'RTSP/1.0 **404 Stream Not Found**',
> [Rebroadcast Plugin]:     code: 404,
> [Rebroadcast Plugin]:     version: 'RTSP/1.0',
> [Rebroadcast Plugin]:     reason: 'Stream'
> [Rebroadcast Plugin]:   }
> [Rebroadcast Plugin]: }
> [Rebroadcast Plugin]: Front Camera restarting prebuffer session in 5 seconds
> [Rebroadcast Plugin]: client error Error: read ECONNRESET
> [Rebroadcast Plugin]:     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
> [Rebroadcast Plugin]:   errno: -104,
> [Rebroadcast Plugin]:   code: 'ECONNRESET',
> [Rebroadcast Plugin]:   syscall: 'read'
> [Rebroadcast Plugin]: }
> [Rebroadcast Plugin]: rebroadcast error Error: read ECONNRESET
> [Rebroadcast Plugin]:     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
> [Rebroadcast Plugin]:   errno: -104,
> [Rebroadcast Plugin]:   code: 'ECONNRESET',
> [Rebroadcast Plugin]:   syscall: 'read'
> [Rebroadcast Plugin]: }

**rtsp://(IP ADDRESS):554/live2:**

> [Rebroadcast Plugin]: Stream 1 prebuffer session started
> [Rebroadcast Plugin]: Stream 1 prebuffer session starting
> stream url rtsp://(IP ADDRESS):554/live2
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  OPTIONS rtsp://(IP ADDRESS):554/live2 RTSP/1.0
> [Rebroadcast Plugin]: CSeq: 0
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 200 OK
> [Rebroadcast Plugin]: CSeq: 0
> [Rebroadcast Plugin]: Public: OPTIONS, DESCRIBE, SETUP, TEARDOWN, PLAY, GET_PARAMETER
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  DESCRIBE rtsp://(IP ADDRESS):554/live2 RTSP/1.0
> [Rebroadcast Plugin]: Accept: application/sdp
> [Rebroadcast Plugin]: CSeq: 1
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 **401 Unauthorized**
> [Rebroadcast Plugin]: CSeq: 1
> [Rebroadcast Plugin]: WWW-Authenticate: Digest realm="uh6h2zo", nonce="AJ5GAEmgHwAIxtt3lw4523i"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  DESCRIBE rtsp://(IP ADDRESS):554/live2 RTSP/1.0
> [Rebroadcast Plugin]: Accept: application/sdp
> [Rebroadcast Plugin]: CSeq: 2
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: Authorization: Digest username="(USERNAME)", realm="uh6h2zo", nonce="AJ5GAEmgHwAIxtt3lw4523i", uri="rtsp://(IP ADDRESS):554/live2", algorithm="MD5", response="1c325c8d25a141653702b668cc1624fd"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 **404 Stream Not Found**
> [Rebroadcast Plugin]: CSeq: 2
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  TEARDOWN rtsp://(IP ADDRESS):554/live2 RTSP/1.0
> [Rebroadcast Plugin]: CSeq: 3
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: Authorization: Digest username="(USERNAME)", realm="uh6h2zo", nonce="AJ5GAEmgHwAIxtt3lw4523i", uri="rtsp://(IP ADDRESS):554/live2", algorithm="MD5", response="277f94296bce8d9479f3d42fe3b2a1c7"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: Stream 1 prebuffer session ended with error M [Error]: RTSP Error: RTSP/1.0 **404 Stream Not Found**
> [Rebroadcast Plugin]:     at t.RtspClient.request (/@scrypted/prebuffer-mixin/main.nodejs.js:1:91847)
> [Rebroadcast Plugin]:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
> [Rebroadcast Plugin]:     at async t.startRtspSession (/@scrypted/prebuffer-mixin/main.nodejs.js:1:144136)
> [Rebroadcast Plugin]:     at async V.startPrebufferSession (/@scrypted/prebuffer-mixin/main.nodejs.js:1:117624) {
> [Rebroadcast Plugin]:   status: {
> [Rebroadcast Plugin]:     line: 'RTSP/1.0 **404 Stream Not Found**',
> [Rebroadcast Plugin]:     code: 404,
> [Rebroadcast Plugin]:     version: 'RTSP/1.0',
> [Rebroadcast Plugin]:     reason: 'Stream'
> [Rebroadcast Plugin]:   }
> [Rebroadcast Plugin]: }
> [Rebroadcast Plugin]: Front Camera restarting prebuffer session in 5 seconds
> [Rebroadcast Plugin]: client error Error: read ECONNRESET
> [Rebroadcast Plugin]:     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
> [Rebroadcast Plugin]:   errno: -104,
> [Rebroadcast Plugin]:   code: 'ECONNRESET',
> [Rebroadcast Plugin]:   syscall: 'read'
> [Rebroadcast Plugin]: }
> [Rebroadcast Plugin]: rebroadcast error Error: read ECONNRESET
> [Rebroadcast Plugin]:     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
> [Rebroadcast Plugin]:   errno: -104,
> [Rebroadcast Plugin]:   code: 'ECONNRESET',
> [Rebroadcast Plugin]:   syscall: 'read'
> [Rebroadcast Plugin]: }

**rtsp://(USERNAME):(PASSWORD)@(IP ADDRESS):554/live2:**

> [Rebroadcast Plugin]: Stream 1 prebuffer session started
> [Rebroadcast Plugin]: Stream 1 prebuffer session starting
> stream url rtsp://(USERNAME):(PASSWORD)@(IP ADDRESS):554/live2
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  OPTIONS rtsp://(IP ADDRESS):554/live2 RTSP/1.0
> [Rebroadcast Plugin]: CSeq: 0
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 200 OK
> [Rebroadcast Plugin]: CSeq: 0
> [Rebroadcast Plugin]: Public: OPTIONS, DESCRIBE, SETUP, TEARDOWN, PLAY, GET_PARAMETER
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  DESCRIBE rtsp://(IP ADDRESS):554/live2 RTSP/1.0
> [Rebroadcast Plugin]: Accept: application/sdp
> [Rebroadcast Plugin]: CSeq: 1
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 **401 Unauthorized**
> [Rebroadcast Plugin]: CSeq: 1
> [Rebroadcast Plugin]: WWW-Authenticate: Digest realm="0x0c2D9", nonce="CAiej6pawBlbttkD3obbfj2"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  DESCRIBE rtsp://(IP ADDRESS):554/live2 RTSP/1.0
> [Rebroadcast Plugin]: Accept: application/sdp
> [Rebroadcast Plugin]: CSeq: 2
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: Authorization: Digest username="(USERNAME)", realm="0x0c2D9", nonce="CAiej6pawBlbttkD3obbfj2", uri="rtsp://(IP ADDRESS):554/live2", algorithm="MD5", response="d84ccc9c4a733c6d3bebf3c0b6e97eeb"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp incoming message
> [Rebroadcast Plugin]:  RTSP/1.0 **404 Stream Not Found**
> [Rebroadcast Plugin]: CSeq: 2
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: rtsp outgoing message
> [Rebroadcast Plugin]:  TEARDOWN rtsp://(IP ADDRESS):554/live2 RTSP/1.0
> [Rebroadcast Plugin]: CSeq: 3
> [Rebroadcast Plugin]: User-Agent: Scrypted
> [Rebroadcast Plugin]: Authorization: Digest username="(USERNAME)", realm="0x0c2D9", nonce="CAiej6pawBlbttkD3obbfj2", uri="rtsp://(IP ADDRESS):554/live2", algorithm="MD5", response="5bf829aa5a75c474a03e7541c344c67e"
> [Rebroadcast Plugin]: 
> [Rebroadcast Plugin]: Stream 1 prebuffer session ended with error M [Error]: RTSP Error: RTSP/1.0 **404 Stream Not Found**
> [Rebroadcast Plugin]:     at t.RtspClient.request (/@scrypted/prebuffer-mixin/main.nodejs.js:1:91847)
> [Rebroadcast Plugin]:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
> [Rebroadcast Plugin]:     at async t.startRtspSession (/@scrypted/prebuffer-mixin/main.nodejs.js:1:144136)
> [Rebroadcast Plugin]:     at async V.startPrebufferSession (/@scrypted/prebuffer-mixin/main.nodejs.js:1:117624) {
> [Rebroadcast Plugin]:   status: {
> [Rebroadcast Plugin]:     line: 'RTSP/1.0 **404 Stream Not Found**',
> [Rebroadcast Plugin]:     code: 404,
> [Rebroadcast Plugin]:     version: 'RTSP/1.0',
> [Rebroadcast Plugin]:     reason: 'Stream'
> [Rebroadcast Plugin]:   }
> [Rebroadcast Plugin]: }
> [Rebroadcast Plugin]: Front Camera restarting prebuffer session in 5 seconds
> [Rebroadcast Plugin]: client error Error: read ECONNRESET
> [Rebroadcast Plugin]:     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
> [Rebroadcast Plugin]:   errno: -104,
> [Rebroadcast Plugin]:   code: 'ECONNRESET',
> [Rebroadcast Plugin]:   syscall: 'read'
> [Rebroadcast Plugin]: }
> [Rebroadcast Plugin]: rebroadcast error Error: read ECONNRESET
> [Rebroadcast Plugin]:     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
> [Rebroadcast Plugin]:   errno: -104,
> [Rebroadcast Plugin]:   code: 'ECONNRESET',
> [Rebroadcast Plugin]:   syscall: 'read'
> [Rebroadcast Plugin]: }

**Reply by MatGriffin** [reply]
I guess this is dead? my 2Cs are in Scrypted but no motion is available.

**Reply by jord0699** [reply]
Im using 2c pro and was able to get snapshot and live feed working in HK via this method but unable to work out how to get it to do secure recording.

based on the other comments, is there any Eufy camera combo that works with scripted?

a little bit of a pain that rtsp is limited to 1080p? id like to gain 2k recording and HK support but doesn't seem like scripted will do this. 

I'm attempting this for the possible HKSV support as I'm currently using a Eufy plugin within Home bridge that works great, just not working with HKSV yet and I always hear good things about scrypted

**Reply by podycust** [reply]
Having a look at the plug in the repo.
When I try to live stream from doorbell cam I get an error.

LivestreamError: Station AES key could not be decrypted! The entire stream is discarded.

**Comment by acerbetti** [comment]
hey folks, I just release my Eufy plugin (very early stage of dev)
https://github.com/caplaz/eufy-security-scrypted

please give it a try and looking forward to collect feedback

**Reply by SmolSoftBoi** [reply]
The architecture looks well thought out, great work. I'd love to see a guide for installing and integrating this with Home Assistant.
